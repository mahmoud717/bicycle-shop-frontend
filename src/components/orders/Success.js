import React from 'react';
import { useHistory } from 'react-router-dom';

const Success = () => {
  const history = useHistory();
  const redirect = () => {
    setTimeout(() => {
      history.push('/');
    }, 5000);
  };
  return (
    <div className="order-success">
      <h1>Your order is created successfully</h1>
      <h3>You will be redirected to the home page in 5 seconds</h3>
      {redirect()}
    </div>
  );
};
export default Success;
