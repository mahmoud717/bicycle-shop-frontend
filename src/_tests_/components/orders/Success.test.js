import React from 'react';

import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Success from '../../../components/orders/Success';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Success />
  </Provider>,
);

describe('Success Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the h1 text of the Success Component', () => {
    const page = component.find('.order-success');
    expect(page).toMatchSnapshot();
  });
});
