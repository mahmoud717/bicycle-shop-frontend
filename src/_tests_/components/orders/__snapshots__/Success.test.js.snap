// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Component renders the h1 text of the Success Component 1`] = `ShallowWrapper {}`;
